CMakeLists.txt

```diff
@@ -99,12 +99,19 @@ setup_build_cache()
 project(rippled)
 
 if(nonunity)
-  set(CMAKE_CXX_FLAGS_DEBUGCLASSIC ${CMAKE_CXX_FLAGS_DEBUG})
-  set(CMAKE_CXX_FLAGS_RELEASECLASSIC ${CMAKE_CXX_FLAGS_RELEASE})
-  set(CMAKE_EXE_LINKER_FLAGS_DEBUGCLASSIC
-    ${CMAKE_EXE_LINKER_FLAGS_DEBUG})
-  set(CMAKE_EXE_LINKER_FLAGS_RELEASECLASSIC
-    ${CMAKE_EXE_LINKER_FLAGS_RELEASE})
+    get_cmake_property(allvars VARIABLES)
+    string(REGEX MATCHALL "[^;]*(DEBUG|RELEASE)[^;]*" matchvars "${allvars}")
+    foreach(var IN LISTS matchvars)
+        string(REGEX REPLACE "(DEBUG|RELEASE)" "\\1CLASSIC" newvar ${var})
+        set(${newvar} ${${var}})
+    endforeach()
+
+    get_cmake_property(allvars CACHE_VARIABLES)
```

Comment:
Normal variables are basically just local, where cache variables persist between runs in `CMakeCache.txt`. I wanted to make sure to get both.

* https://cmake.org/cmake/help/v3.4/command/set.html
* https://cmake.org/cmake/help/v3.4/command/get_cmake_property.html
* https://cmake.org/cmake/help/v3.4/prop_dir/CACHE_VARIABLES.html
* https://cmake.org/cmake/help/v3.4/prop_dir/VARIABLES.html